File: rust/src/models/product_family.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::product_family_id::ProductFamilyId;

#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct ProductFamily {
    pub id: ProductFamilyId,

    pub name: String,
}

impl ProductFamily {
    pub fn new(id: ProductFamilyId, name: String) -> Self {
        Self { id, name }
    }
}

File: typescript/src/models/productFamily.ts
// this file is @generated
import { type ProductFamilyId, ProductFamilyIdSerializer } from "./productFamilyId";

export interface ProductFamily {
  id: ProductFamilyId;

  name: string;
}

export const ProductFamilySerializer = {
  _fromJsonObject(object: any): ProductFamily {
    return {
      id: ProductFamilyIdSerializer._fromJsonObject(object["id"]),
      name: object["name"],
    };
  },

  _toJsonObject(self: ProductFamily): any {
    return {
      id: ProductFamilyIdSerializer._toJsonObject(self.id),
      name: self.name,
    };
  },
};
